WordPress/gutenberg
# Reposition navigation selector

## What?


This is a spin off from https://github.com/WordPress/gutenberg/pull/44534.

## Why?


Because it is a secondary action to menu editing, and to accommodate the presence of a full list view in the navigation block inspector, the navigation selector's impact has to be reduced.

## How?


- Creates a custom header for the menu panel in the inspector. 
- Adds a more menu to this panel header.
- Moves the navigation selector into this more menu when experiment active.


## Testing Instructions





1. Switch to this PR
2. Go to Gutenberg > Experiments
3. Toggle on "Test a new "off canvas" editor for navigation block using the block inspector and a tree view of the current menu"
4. Save
5. Create a post
6. Add a navigation block
7. Create a block menu with sample items if one is not selected (it has to be a menu)
8. Select the block
9. Notice the new list view in the block inspector
10. Notice the dotted "more menu" in the panel that contains the list view, this is where the menu selector is now.

## Screenshots or screencast 


https://user-images.githubusercontent.com/107534/199258779-af083416-e0c1-4169-b3bd-233a4adfccd3.mp4
